spec/ysh-slice-range.test.sh

```diff
@@ -101,6 +101,26 @@ var b = a[:]
 (List)   [1, 2, 3]
 ## END
 
+#### Iterate over range
+for i in (1:5) {
+    echo $[i]
+}
+## STDOUT:
+1
+2
+3
+4
+## END
+
+#### Loops over bogus ranges terminate
+# Regression test for bug found during dev. Loops over backwards ranges should
+# terminate immediately.
```

Comment:
This is consistent with how python handles `for i in range(5, 1): print(i)`. We could also treat this as an error when we evaluate range expressions. Not sure which is better...

Reply:
OK great, I think Python's semantics makes sense

(could be convinced otherwise, but Python is a good default ... I think that range bounds can be computed, and empty ranges make sense, like empty slices)